### ISSUE TITLE
[http-client-csharp] post processor is not treating the customization code as root node

### ISSUE THREAD

**Issue by ArcturusZhang** [microsoft/typespec#5441]
Customization code should be treated as root node as well, now we are not, therefore changing the accessibility via customization code will not work when we have the postprocessor on because post processor will ignore everything in customization code and internalize and/or remove it if the model is not used.
